src/Raven.Client/Json/Serialization/JsonNet/Internal/BlittableJsonConverterBase.cs

```diff
@@ -0,0 +1,285 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Dynamic;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Raven.Client.Documents.Conventions;
+using Raven.Client.Util;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Client.Json.Serialization.JsonNet.Internal
+{
+    internal abstract class BlittableJsonConverterBase : IBlittableJsonConverterBase
+    {
+        protected readonly ISerializationConventions Conventions;
+
+        protected BlittableJsonConverterBase(ISerializationConventions conventions)
+        {
+            Conventions = conventions ?? throw new ArgumentNullException(nameof(conventions));
+        }
+
+        public void PopulateEntity(object entity, BlittableJsonReaderObject json)
+        {
+            var jsonSerializer = Conventions.CreateSerializer();
+            PopulateEntity(entity, json, jsonSerializer);
+        }
+
+        public void PopulateEntity(object entity, BlittableJsonReaderObject json, IJsonSerializer jsonSerializer)
+        {
+            if (entity == null)
+                throw new ArgumentNullException(nameof(entity));
+            if (json == null)
+                throw new ArgumentNullException(nameof(json));
+            if (jsonSerializer == null)
+                throw new ArgumentNullException(nameof(jsonSerializer));
+
+            var serializer = (JsonNetJsonSerializer)jsonSerializer;
+            var old = serializer.ObjectCreationHandling;
+            serializer.ObjectCreationHandling = ObjectCreationHandling.Replace;
+
+            try
+            {
+                using (var reader = new BlittableJsonReader())
+                {
+                    reader.Initialize(json);
+
+                    serializer.Populate(reader, entity);
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new InvalidOperationException($"Could not populate entity.", ex);
+            }
+            finally
+            {
+                serializer.ObjectCreationHandling = old;
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Remove this, the method is too big

Reply:
Ok, just copied what we had in previous one.